src/main/java/org/opensearch/ad/rest/RestStatsAnomalyDetectorAction.java

```diff
@@ -11,64 +11,119 @@
+    private SDKRestClient sdkRestClient;
+    private Settings settings;
+    @Inject
+    private DiscoveryNodeFilterer discoveryNodeFilterer;
+
+    public RestStatsAnomalyDetectorAction(ExtensionsRunner extensionsRunner, SDKRestClient sdkRestClient) {
+        this.sdkRestClient = sdkRestClient;
+        this.settings = extensionsRunner.getEnvironmentSettings();
     }
 
-    @Override
     public String getName() {
         return STATS_ANOMALY_DETECTOR_ACTION;
     }
 
     @Override
-    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
+    public List<RouteHandler> routeHandlers() {
+        return ImmutableList
+            .of(
+                new RouteHandler(RestRequest.Method.GET, String.format(Locale.ROOT, "/{%s}/%s", NODE_ID, "stats"), handleRequest),
+                new RouteHandler(
+                    RestRequest.Method.GET,
+                    String.format(Locale.ROOT, "/{%s}/%s/{%s}", NODE_ID, "stats", STAT),
+                    handleRequest
+                ),
+                new RouteHandler(RestRequest.Method.GET, "/stats", handleRequest),
+                new RouteHandler(RestRequest.Method.GET, String.format(Locale.ROOT, "/%s/{%s}", "stats", STAT), handleRequest)
+            );
+    }
+
+    private Function<RestRequest, ExtensionRestResponse> handleRequest = (request) -> {
+        try {
+            return prepareRequest(request);
+        } catch (Exception e) {
+            // TODO: handle the AD-specific exceptions separately
+            return exceptionalRequest(request, e);
+        }
+    };
+
+    protected ExtensionRestResponse prepareRequest(RestRequest request) throws IOException {
         if (!EnabledSetting.isADPluginEnabled()) {
             throw new IllegalStateException(CommonErrorMessages.DISABLED_ERR_MSG);
         }
         ADStatsRequest adStatsRequest = getRequest(request);
-        return channel -> client.execute(StatsAnomalyDetectorAction.INSTANCE, adStatsRequest, new RestToXContentListener<>(channel));
+        CompletableFuture<StatsAnomalyDetectorResponse> statsFutureResponse = new CompletableFuture<>();
+        sdkRestClient
+            .execute(
+                StatsAnomalyDetectorAction.INSTANCE,
+                adStatsRequest,
+                ActionListener.wrap(response -> statsFutureResponse.complete(response), ex -> statsFutureResponse.completeExceptionally(ex))
+            );
+
+        StatsAnomalyDetectorResponse statsAnomalyDetectorResponse = statsFutureResponse
+            .orTimeout(AnomalyDetectorSettings.REQUEST_TIMEOUT.get(settings).getMillis(), TimeUnit.MILLISECONDS)
+            .join();
+
+        XContentBuilder statsAnomalyDetectorResponseBuilder = statsAnomalyDetectorResponse
+            .toXContent(JsonXContent.contentBuilder(), ToXContent.EMPTY_PARAMS);
+        ExtensionRestResponse response = new ExtensionRestResponse(request, RestStatus.OK, statsAnomalyDetectorResponseBuilder);
+
+        return response;
```

Comment:
(Style) Why not just `return new ExtensionRestResponse(...)` here instead of creating a new object and returning it on the next line?  Yeah, the compiler optimizes this out, just reads better to me. :)